### PR TITLE
fix: Making classes accessible to windows via uno.ui.toolkit

### PR DESCRIPTION
**Pull request by nickrandolph**
## PR Type

What kind of change does this PR introduce?
- Feature

## What is the current behavior?

HR typemapping and elementmetadataupdate attribute only accessible via Uno.WinUI (meaning windows can't access it)

## What is the new behavior?

HR typemapping and elementmetadataupdate attribute are in Uno.UI.Toolkit for Windows, meaning that Windows can reference the types

### <samp>🤖 Generated by Copilot at 3b8b3a4</samp>

This pull request updates the Uno.UI.Toolkit project to use some classes from the Uno.UI project that are needed for the hot reload feature. It also applies the new file-scoped namespace syntax from C# 10 to the ElementMetadataUpdateHandlerAttribute and TypeMappings classes, to simplify the code and reduce the indentation level.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/Helpers/TypeMappings.cs [hunk 1]
```diff
@@ -4,134 +4,133 @@
...
+	public static Task WaitForMappingsToResume()
+		=> _mappingsPaused is not null ? _mappingsPaused.Task : Task.CompletedTask;
+
+	/// <summary>
+	/// Pause the collection of type mappings.
+	/// Internally the type mappings are still collected but will only be
+	/// applied to the mapping dictionaries after Resume is called
+	/// </summary>
+	public static void Pause()
+		=> _mappingsPaused ??= new TaskCompletionSource<object>();
```
Comment (Youssef1313) [inline src/Uno.UI/Helpers/TypeMappings.cs:117]:
```suggestion
		=> _mappingsPaused ??= new TaskCompletionSource<bool>();
```

File: src/Uno.UI/Helpers/TypeMappings.cs [hunk 2]
```diff
@@ -4,134 +4,133 @@
...
-			_mappingsPaused = null;
-			if (completion is not null)
-			{
-				MappedTypeToOrignalTypeMapings = AllMappedTypeToOrignalTypeMapings.ToDictionary(x => x.Key, x => x.Value);
-				OriginalTypeToMappedType = AllOriginalTypeToMappedType.ToDictionary(x => x.Key, x => x.Value);
-				completion.TrySetResult();
-			}
+			MappedTypeToOrignalTypeMapings = AllMappedTypeToOrignalTypeMapings.ToDictionary(x => x.Key, x => x.Value);
+			OriginalTypeToMappedType = AllOriginalTypeToMappedType.ToDictionary(x => x.Key, x => x.Value);
+			completion.TrySetResult(new object());
```
Comment (Youssef1313) [inline src/Uno.UI/Helpers/TypeMappings.cs:133]:
```suggestion
			completion.TrySetResult(true);
```

File: src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.Agent.cs [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+			{
+				if (this.Log().IsEnabled(LogLevel.Warning))
+				{
+					this.Log().Trace($"Runtime does not support Hot Reload (Unable to find type {HotReloadAgent.MetadataUpdaterType})");
+				}
+			}
+			return Array.Empty<string>();
+		}
+
+#if __WASM__ || __SKIA__
```
Comment (jeromelaban) [inline src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.Agent.cs:123]: This has been changed in a merged PR, it's probably best if you re-do this change to align.

Comment (jeromelaban) [reply src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.Agent.cs:123]: Actually, the update was lost somehow... it's very strange, I'll change it again in a separate PR.

Comment (nickrandolph) [reply src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.Agent.cs]: rebased and picked up latest changes

File: src/Uno.UI.RemoteControl/HotReload/WindowExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+/// <summary>
+/// Extension methods for the Window class
+/// </summary>
+public static class WindowExtensions
+{
+	/// <summary>
+	/// Enables the UI Update cycle of HotReload to be handled by Uno
+	/// </summary>
+	/// <param name="window">The window of the application where UI updates will be applied</param>
+	public static void EnableHotReload(this Window window) => ClientHotReloadProcessor.CurrentWindow = window;
```
Comment (dr1rrb) [inline src/Uno.UI.RemoteControl/HotReload/WindowExtensions.cs:15]: This is not multi-window compatible :) (Mot a blocker IMHO)

**Review by dr1rrb — APPROVED** [review]

File: src/Uno.UI.RemoteControl/HotReload/WindowExtensions.cs (hunk 4 above)
Comment (nickrandolph) [reply src/Uno.UI.RemoteControl/HotReload/WindowExtensions.cs:15]: And in net8 we'll generate this using interceptors

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/5.0

**Comment by jeromelaban** [comment]
https://github.com/Mergifyio backport release/stable/5.0

**Pull request merged**